common/src/main/java/io/netty/util/internal/PlatformDependent0.java

```diff
@@ -311,6 +318,37 @@ static void copyMemory(Object src, long srcOffset, Object dst, long dstOffset, l
         }
     }
 
+    static boolean equals(byte[] bytes1, int startPos1, byte[] bytes2, int startPos2) {
```

Comment:
We also need `bytes1Length` and `bytes2Length`.  I don't think we can use `bytes1.length` and `bytes2.length` because we allow sharing of "slices" or "sub regions" of arrays.